codegenerator/cli/templates/static/codegen/src/EventModLookup.res

```diff
@@ -0,0 +1,109 @@
+open Belt
+type eventMod = module(Types.InternalEvent)
+
+type errorKind = WildcardSighashCollision | Duplicate
+type eventError = {eventMod: eventMod, errorKind: errorKind}
+module ContractEventMods = {
+  type t = {
+    all: array<eventMod>,
+    byContractName: dict<eventMod>,
+  }
+
+  let empty = () => {
+    all: [],
+    byContractName: Js.Dict.empty(),
+  }
+
+  let isWildcard = (eventMod: eventMod) => {
+    let module(Event) = eventMod
+    Event.handlerRegister->Types.Handlers.Register.getEventOptions->(v => v.Types.Handlers.wildcard)
+  }
+
+  let hasWildcardCollision = (eventModA, eventModB) => {
+    eventModA->isWildcard || eventModB->isWildcard
+  }
+
+  let set = ({all, byContractName}: t, eventMod: eventMod) => {
+    let module(Event) = eventMod
+    switch byContractName->Js.Dict.get(Event.contractName) {
```

Comment:
A tiny thing, but there's `Utils.Dict.dangerouslyGetNonOption`, which doesn't have an unnecessary runtime overhead.

Reply:
Ok cool, I think it makes sense to improve for the get function but these values will only be set at startup